agent/pkg/client/credentials_test.go

```diff
@@ -0,0 +1,362 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+
+	"golang.org/x/oauth2"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type mockServiceAccountInterface struct {
+	token      string
+	expiration time.Time
+	err        error
+
+	calls              int
+	serviceAccountName string
+	audiences          []string
+}
+
+var _ serviceAccountInterface = &mockServiceAccountInterface{}
```

Comment:
nit: we don't need this line since we're using `mockServiceAccountInterface` as `serviceAccountInterface` later L67, so incorrectly implementation of the interface will be surfaced anyway.

Reply:
I always make these declarations immediately before the implementation methods of the given interface.